src/App.tsx

```diff
@@ -8,7 +7,56 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { SortType } from './types/sortType';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [sortType, setSortType] = useState(SortType.All);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [query, setQuery] = useState('');
+  const [activeTodoId, setActiveTodoId] = useState(0);
+
+  const activeTodo = todos.find(todo => todo.id === activeTodoId);
+
+  const getTodosList = useCallback(async (promise: Promise<Todo[]>) => {
+    try {
+      setTodos(await promise);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  let visibleTodos = todos;
+
+  switch (sortType) {
+    case SortType.Active:
+      visibleTodos = visibleTodos.filter(todo => !todo.completed);
+      break;
+
+    case SortType.Completed:
+      visibleTodos = visibleTodos.filter(todo => todo.completed);
+      break;
+
+    default:
+      break;
+  }
+
+  visibleTodos = visibleTodos
+    .filter(todo => todo.title.toLocaleLowerCase()
+      .includes(query.toLocaleLowerCase()));
+
+  const errorTodoList = todos.length > 0 && !isError && !isLoading;
+
+  useEffect(() => {
+    getTodosList(getTodos());
+  }, []);
```

Comment:
I dont really understand why you pass getTodos function as a parameter to getTodosList. You could simply wait for the result of getTodos inside getTodosList. Check my suggestion please